### ISSUE TITLE
Spring failing to connect behind corporate proxy for the out-of-process deployer

### ISSUE THREAD

**Issue by joriscaloud** [spring-cloud/spring-cloud-dataflow#420]
Hi,

We are failing to deploy the basic ticktock stream on Spring cloud dataflow.

We are working behind a corporate proxy  and it seems that the repo system doesn't take any informations from our `dataflow-server.yml ` (it also fails by lauching it with -dHttp parameters)

`server:
  port: 9393
management:
  contextPath: /management
  security:
    enabled: false
info:
  app:
    name: "@project.artifactId@"
    description: "@project.description@"
    version: "@project.version@"
security:
  basic:
    enabled: false

spring:
  application:
    name: spring-cloud-dataflow-server
  cloud:
    config:
      uri: http://localhost:8888
  datasource:
    url: jdbc:h2:tcp://localhost:19092/mem:dataflow
    username: sa
    password:
    driverClassName: org.h2.Driver
  task:
    repo:
      initialize: true
aether:
  proxy:
    host: proxy.corp.fr
    port: XXXX`

Our ticktock logs : 

[stderr_0.txt](https://github.com/spring-cloud/spring-cloud-dataflow/files/146432/stderr_0.txt)
[stdout_0.txt](https://github.com/spring-cloud/spring-cloud-dataflow/files/146431/stdout_0.txt)

Has  someone  already experienced this problem or has any  suggestions on this issue ? 

Many thanks

**Comment by ilayaperumalg** [comment]
@joriscaloud we recently pushed a fix to support aether proxy settings for the module launcher: https://github.com/spring-cloud/spring-cloud-stream/commit/b98d91da394d4d98dbf08e3cb1bd30455825c76d.

I want to confirm if you are using the latest module launcher that includes this fix. If so, this should be bug. Could you share some more debug level log (especially for the ModuleLauncher side)?

**Comment by joriscaloud** [comment]
Hi @ilayaperumalg ,

Yes we saw this fix, we are currently using it (we cloned the yesterday's spring-cloud-dataflow version).

Can you  specify  the way to retrieve some debug level log that can help you (for the ModuleLauncher side) please ? thanks

**Comment by ilayaperumalg** [comment]
You can do it this way (for the in-process deployer):

```
java -jar spring-cloud-dataflow-server-local/target/spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar --logging.level.org.eclipse.aether=DEBUG --deployer.local.out-of-process=false
```

**Comment by joriscaloud** [comment]
Hi again,

please find attached the debug that we've got by launching your command line. Please tell us if this is what you expected. Thanks
[debug_mode.txt](https://github.com/spring-cloud/spring-cloud-dataflow/files/146496/debug_mode.txt)

**Comment by joriscaloud** [comment]
It works now, it seems that we didn't have the deployer.local.out-of-process=false in our config.

We still don't know what is the effect of this line (out of process ?), can you add it to the documentation please ? 
Thank you for your help !

**Comment by ilayaperumalg** [comment]
The flag `deployer.local.out-of-process=false` indicates the `local` deployer to use **in-process** deployer (the **out-of-process** deployer is the default). 

It looks like the aether proxy settings work in case of `in-process` but not for `out-of-process` deployer. We will investigate and update.

@joriscaloud Thanks for reporting and trying out!

**Comment by venkiram** [comment]
I still face the issue with "Cloud data flow" working behind proxy. I took the latest source (from git) and Maven-ed it...it went fine. But when I try to create & deploy a stream it errors out saying  - 

Command failed org.springframework.cloud.dataflow.rest.client.DataFlowClientException: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact org.springframework.cloud.stream.module:log-sink:jar:exec:1.0.0.BUILD-SNAPSHOT from/to repository 1 (https://repo.spring.io/libs-snapshot): repo.spring.io

Also tried the deployer.local.out-of-process=false option as given in this thread, still no luck. Let me know.

**Comment by ilayaperumalg** [comment]
@venkiram It would be helpful to see the debug log to get the root cause of the failure in your case and to confirm if the proxy properties are correctly applied.

**Comment by venkiram** [comment]
Thanks for the quick response. How will I check if the proxy properties are applied correctly - the proxy settings works fine in all other places (like wget, Maven etc.) and I'm passing it in the command line as well like -Dhttp.useProxy=true -Dhttps.proxyHost=x.x.x.x -Dhttps.proxyPort=81 -Dhttp.nonProxyHosts=127.0.0.1

**Comment by ilayaperumalg** [comment]
also, check if you run into the problem mentioned here: https://github.com/spring-cloud/spring-cloud-dataflow/pull/422

**Comment by Mwea** [comment]
@venkiram Please be sure to set your [dataflow-server.yml](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-deployer-local/src/main/resources/dataflow-server.yml) 

by adding this at the end of your file, just before the commented lines : 
`aether:
  proxy:
    host: your.proxy.com
    port: XXX`

**Comment by venkiram** [comment]
Yes, putting the code fix in #422 and setting the aether proxy, fixes the proxy issue and I'm able to create stream now. Thanks for the support.
